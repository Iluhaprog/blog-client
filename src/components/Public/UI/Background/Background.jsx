import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const BackgroundBox = styled.div`
  position: relative;
  max-height: ${({maxHeight}) => maxHeight}px;
  width: 100%;
  height: 100vh;
`;

const ImageBox = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  
  img {
    width: 100%;
    height: 100%;
    object-position: center;
    object-fit: cover;
  }
`;

const ContentBox = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const Background = ({
  maxHeight,
  src,
  alt,
  children,
}) => {
  return (
    <BackgroundBox maxHeight={maxHeight}>
      <ImageBox>
        <img src={src} alt={alt}/>
      </ImageBox>
      <ContentBox>
        {children}
      </ContentBox>
    </BackgroundBox>
  );
};

Background.propTypes = {
  maxHeight: PropTypes.number,
  src: PropTypes.string,
  alt: PropTypes.string,
  children: PropTypes.node,
};

export {Background};
