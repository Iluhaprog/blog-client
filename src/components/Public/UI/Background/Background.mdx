---
name: Background
route: /background
---

import {Playground, Props} from 'docz';
import {Card} from '../Card';
import {Background} from './Background';

<Props of={Background} />

## Basic usage
<Playground>
  <Background
    maxHeight={200}
    src={'https://images.ctfassets.net/hrltx12pl8hq/4plHDVeTkWuFMihxQnzBSb/aea2f06d675c3d710d095306e377382f/shutterstock_554314555_copy.jpg'}
    alt={'bg'}
  >
    <Card maxWidth={100} maxHeight={100}>
      Child
    </Card>
  </Background>
</Playground>
