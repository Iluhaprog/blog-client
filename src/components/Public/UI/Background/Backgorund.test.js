import React from 'react';
import {render} from '@testing-library/react';
import {Background} from './Background';

test('Should render Background component', () => {
  const {asFragment} = render(
    <Background
      maxHeight={200}
      src={'https://static.toiimg.com/photo/72975551.cms'}
      alt={'img'}
    >
      Hello
    </Background>
  );
  expect(asFragment()).toMatchSnapshot();
});
