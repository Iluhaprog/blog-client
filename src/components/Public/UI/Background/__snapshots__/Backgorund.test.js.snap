// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Background component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ llrYKs"
  >
    <div
      class="sc-gsTCUz kazNgE"
    >
      <img
        alt="img"
        src="https://static.toiimg.com/photo/72975551.cms"
      />
    </div>
    <div
      class="sc-dlfnbm iXCpsF"
    >
      Hello
    </div>
  </div>
</DocumentFragment>
`;
